src/main/java/org/spongepowered/common/data/processor/data/entity/IgniteableDataProcessor.java

```diff
@@ -0,0 +1,147 @@
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<IgniteableData> fill(DataHolder dataHolder, IgniteableData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final IgniteableData merged = overlap.merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+            manipulator.set(Keys.FIRE_DAMAGE_DELAY, merged.fireDelay().get()).set(Keys.FIRE_TICKS, merged.fireTicks().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<IgniteableData> fill(DataContainer container, IgniteableData igniteableData) {
+        igniteableData.set(Keys.FIRE_TICKS, getData(container, Keys.FIRE_TICKS));
+        igniteableData.set(Keys.FIRE_DAMAGE_DELAY, getData(container, Keys.FIRE_DAMAGE_DELAY));
+        return Optional.of(igniteableData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, IgniteableData manipulator, MergeFunction function) {
+        final int oldFireDamageDelay = manipulator.fireDelay().get();
+        final int oldFireTicks = manipulator.fireTicks().get();
+        final IgniteableData merged =
+                checkNotNull(function, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+        manipulator.set(Keys.FIRE_DAMAGE_DELAY, merged.fireDelay().get());
+        manipulator.set(Keys.FIRE_TICKS, merged.fireTicks().get());
+        final int fireDamageDelay = merged.fireDelay().get();
+        final int fireTicks = merged.fireTicks().get();
+        if (this.supports(dataHolder)) {
+            net.minecraft.entity.Entity entity = (net.minecraft.entity.Entity) dataHolder;
+            entity.fire = fireTicks;
+            entity.fireResistance = fireDamageDelay;
+            Collection<ImmutableValue<?>> successful = new ArrayList<ImmutableValue<?>>();
+            Collection<ImmutableValue<?>> replaced = new ArrayList<ImmutableValue<?>>();
+            successful.add(new ImmutableSpongeValue<Integer>(Keys.FIRE_TICKS, oldFireTicks));
+            replaced.add(new ImmutableSpongeValue<Integer>(Keys.FIRE_TICKS, fireTicks));
+            successful.add(new ImmutableSpongeValue<Integer>(Keys.FIRE_DAMAGE_DELAY, oldFireDamageDelay));
+            replaced.add(new ImmutableSpongeValue<Integer>(Keys.FIRE_DAMAGE_DELAY, fireDamageDelay));
+            return DataTransactionBuilder.successReplaceResult(successful, replaced);
+        }
+        return DataTransactionBuilder.failResult(manipulator.getValues());
+    }
+
+    @Override
+    public Optional<ImmutableIgniteableData> with(Key<? extends BaseValue<?>> key, Object value, ImmutableIgniteableData immutable) {
+        if (key.equals(Keys.FIRE_DAMAGE_DELAY)) {
+            return Optional.<ImmutableIgniteableData>of(ImmutableDataCachingUtil.getManipulator(ImmutableSpongeIgniteableData.class, value));
```

Comment:
This isn't how the caching data works.... You're not providing enough information for the constructor to be called correctly, let alone the fact that this isn't the best type of data to cache.